fix: product category

jonathanwlopes opened the pull request:
Hey guys,

In this pullrequest I make a fix so that the complete product categories are returned in the schema.org standard.

Return before PR:
![current-category](https://github.com/deco-cx/apps/assets/65980571/2bd2ea9a-26a7-49b7-8f7c-a99e40ee4791)

Return after PR:
![fixed-category](https://github.com/deco-cx/apps/assets/65980571/b06b0f54-9e64-4548-9482-04648ef4be76)

### **A big point of attention**

As previously, only one category was returned in this property,
Many developers may have used this category without DEFAULT_CATEGORY_SEPARATOR handling in the frontend.

But according to schema.org documentation,
I didn't find another product property that can receive all product categories

This correction is necessary for the use cases of sending categories in analytics events

See what it was like before PR:
![current-analytics](https://github.com/deco-cx/apps/assets/65980571/2bbe6041-f842-4cf5-b040-43f78854ef7b)

See it's now:
![fixed-analytics](https://github.com/deco-cx/apps/assets/65980571/581ff6b8-05eb-4d78-8162-4dad2aa0b151)

Furthermore, I ask for help to understand the above point of attention and get this data to manipulate it in the frontend.

devartes commented:
Product
A Schema.org Type
[Thing](https://schema.org/Thing) > [Product](https://schema.org/Product)

> Any offered product or service. For example: a pair of shoes; a concert ticket; the rental of a car; a haircut; or an episode of a TV show streamed online.

Using > to separate is not the ideal way
```
product: {
type: "Product",
category: "Cozinha>Banheiro>Ventosa>Por Cores Cozinha>Por Cores Banheiro>Ventosas Banheiro>
Ganchos>Cozinha Preta>Banheiro Preto>Gancho com Ventosa>Gancho>Por Cores Ventosas>
Ventosa Preto Fosco>Ventosa Banheiro Preto Fosco>Inspirações>Aniversário Ela Decora",
...
```
[Product](https://schema.org/Product) from [schema.org](https://schema.org/)
A list would be better than by >

    {
      "@context": "https://schema.org/",
      "@type": "Product",
      "category": [
        "Cozinha",
        "Banheiro",
        "Ventosa",
        "Por Cores Cozinha",
        "Por Cores Banheiro",
        "Ventosas Banheiro",
        "Ganchos",
        "Cozinha Preta",
        "Banheiro Preto",
        "Gancho com Ventosa",
        "Gancho",
        "Por Cores Ventosas",
        "Ventosa Preto Fosco",
        "Ventosa Banheiro Preto Fosco",
        "Inspirações",
        "Aniversário Ela Decora"
      ],
      "productID": "2001300",
      "url": "http://localhost:8000/gancho-multiuso-com-ventosa-preto-fosco-11-5cm/p?skuId=2001300",
      "name": "Gancho de Parede Multiuso com Ventosa Preto Fosco 11,5cm Design Moderno",
      ...
    }

jonathanwlopes commented:
Hello everyone, good afternoon!

I am very grateful for the recommendations above,

Based on my reading and study, I really liked the proposal:
"Proposal 2: Utilize Additional Property for Extra Categories" made by @matheusgr.

With that, I reverted the 'category' property to how it was before the pull request, and added the 'Additional Categories' property under Additional Property.

![image](https://github.com/deco-cx/apps/assets/65980571/a20e0442-e81a-4889-980f-8cb9c305666d)

If you could analyze this new proposal to receive all product categories, that would be great.

I appreciate the attention so far.

Thank you again, @matheusgr @devartes

guitavano commented on commerce/types.ts:
```diff
@@ -86,7 +86,7 @@ export interface PropertyValue extends Omit<Thing, "@type"> {
    * - Use '.' (Unicode 'FULL STOP' (U+002E)) rather than ',' to indicate a decimal point. Avoid using these symbols as a readability separator.
    */
-  value?: string;
+  value?: string | PropertyValue[];
```
It is not at schema.org, right? schema.org/PropertyValue
